sinks/sinks_service.go

```diff
@@ -62,71 +65,191 @@ func (svc sinkService) CreateSink(ctx context.Context, token string, sink Sink)
+	var err error
+	if sink.Config != nil {
+		encodeMetadata, err := configSvc.Authentication.EncodeInformation("object", sink.Config)
+		if err != nil {
+			svc.logger.Error("error on parsing encrypted config in data")
+			return sink, err
 		}
-		return key, newValue
-	})
+		sink.Config = encodeMetadata.(types.Metadata)
+	}
 	if sink.ConfigData != "" {
-		sinkBE := backend.GetBackend(sink.Backend)
-		if sinkBE == nil {
-			return sink, errors.New("backend cannot be nil")
-		}
-		sink.ConfigData, err = sinkBE.ConfigToFormat(sink.Format, sink.Config)
+		encodeMetadata, err := configSvc.Authentication.EncodeInformation("yaml", sink.ConfigData)
 		if err != nil {
 			svc.logger.Error("error on parsing encrypted config in data")
 			return sink, err
 		}
+		sink.ConfigData = encodeMetadata.(string)
 	}
-
 	return sink, err
 }
 
-func (svc sinkService) decryptMetadata(sink Sink) (Sink, error) {
+func (svc sinkService) ViewAuthenticationType(ctx context.Context, token string, key string) (authentication_type.AuthenticationTypeConfig, error) {
+	_, err := svc.identify(token)
+	if err != nil {
+		return authentication_type.AuthenticationTypeConfig{}, err
+	}
+
+	value, ok := authentication_type.GetAuthType(key)
+	if !ok {
+		return authentication_type.AuthenticationTypeConfig{}, errors.New("invalid authentication type given name")
+	}
+	return value.Metadata(), nil
+}
+
+func (svc sinkService) ListAuthenticationTypes(ctx context.Context, token string) ([]authentication_type.AuthenticationTypeConfig, error) {
+	_, err := svc.identify(token)
+	if err != nil {
+		return nil, err
+	}
+
+	value := authentication_type.GetList()
+
+	return value, nil
+}
+
+func (svc sinkService) decryptMetadata(configSvc Configuration, sink Sink) (Sink, error) {
 	var err error
-	sink.Config.FilterMap(func(key string) bool {
-		return key == backend.ConfigFeatureTypePassword
-	}, func(key string, value interface{}) (string, interface{}) {
-		newValue, err2 := svc.passwordService.DecodePassword(value.(string))
-		if err2 != nil {
-			err = err2
-			return key, value
+	if sink.Config != nil {
+		decodeMetadata, err := configSvc.Authentication.DecodeInformation("object", sink.Config)
+		if err != nil {
+			svc.logger.Error("error on parsing encrypted config in data")
+			return sink, err
 		}
-		return key, newValue
-	})
+		sink.Config = decodeMetadata.(types.Metadata)
+	}
 	if sink.ConfigData != "" {
-		sinkBE := backend.GetBackend(sink.Backend)
-		if sinkBE == nil {
-			return sink, errors.New("backend cannot be nil")
-		}
-		sink.ConfigData, err = sinkBE.ConfigToFormat(sink.Format, sink.Config)
+		decodeMetadata, err := configSvc.Authentication.DecodeInformation("yaml", sink.ConfigData)
 		if err != nil {
 			svc.logger.Error("error on parsing encrypted config in data")
 			return sink, err
 		}
+		sink.ConfigData = decodeMetadata.(string)
 	}
 	return sink, err
 }
 
+func (svc sinkService) UpdateSinkInternal(ctx context.Context, sink Sink) (Sink, error) {
```

Comment:
I believe this is only called during migration - if so, let's document that in the function comments here (and/or in the interface def)